numba/typed/typedobjectutils.py

```diff
@@ -197,3 +197,78 @@ def build_wrapper(fn):
     builder.ret(context.get_constant(types.int32, -1))
 
     return equal_fn
+
+
+def _get_primitive_equal(context, module, datamodel, element_type):
```

Comment:
I'd change `element_type` to something more meaningful. It is not an element type but rather a suffix: `_dict_key`. In fact, there's only one place this function is called.